### DISCUSSION TITLE
Callouts in nested blocks

### DISCUSSION THREAD

**Discussion by neeldhara** [quarto-dev/quarto-cli#3607] [Q&A]
When inserting a callout using the standard callout syntax like so:

```
:::{.callout-tip}
Foo bar.
:::
```

indented inside, say, a numbered list --- it simply does not work :( Quarto seems to render the syntax as raw text. Is this by design or am I doing something wrong?

**Source:**
![CleanShot 2022-12-09 at 01 17 54](https://user-images.githubusercontent.com/15953732/206553123-152fb957-bb80-4508-ae8f-6a1028bd5c31.png)

**Rendered (HTML):**
![CleanShot 2022-12-09 at 01 18 03](https://user-images.githubusercontent.com/15953732/206553119-f5ef076f-0c79-4fdc-9894-e0082abcc57d.png)

**Comment by dragonstyle** [comment]
Markdown spacing / whitespace can be finicky. I was able to place a callout in an ordered list like so:

```markdown

1. Item 1
2. Item 2

   :::{.callout-note}
   ## Sample Callout

   This is a sample callout

   :::
   
3. Item 3

```

HTH!

**Reply by mcanouil** [reply]
In addition, when nesting in Quarto/Pandoc, the nested element must be aligned to the first character.
In @dragonstyle example, this means the callout first colon ":" has to be aligned with the "I" of item.

RMarkdown was using indent level, this is where the difference can hurt when switching to Quarto.

**Reply by dragonstyle** [reply]
> RMarkdown was using indent level, this is where the difference can hurt when switching to Quarto.

I didn't know this- ty for adding this!

**Reply by mcanouil** [reply]
I faced this issue and J.J. enlighted me on this difference^^

See #948

**Reply by mcanouil** [reply]
I did not notice this for fenced divs but they behave as code cells, i.e., blank lines around them are required to ensure they are parsed properly when nested in list.

To summarise:

```md
---
format: html
---

## Good

1. Item 1
2. Item 2

   :::{.callout-note}
   ## Sample Callout
 
   This is a sample callout
 
   :::

3. Item 3

## Bad (wrong alignment)

1. Item 1
2. Item 2

  :::{.callout-note}
  ## Sample Callout

  This is a sample callout

  :::

3. Item 3

## Bad (no blank lines around fenced divs)

1. Item 1
2. Item 2
   :::{.callout-note}
   ## Sample Callout

   This is a sample callout

   :::
3. Item 3

## Bad (wrong alignment and no blank lines around fenced divs)

1. Item 1
2. Item 2
  :::{.callout-note}
  ## Sample Callout

  This is a sample callout

  :::
3. Item 3
```
